data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -59,24 +67,31 @@ void parseS3Object(final S3ObjectReference s3ObjectPointer) throws IOException {
 
         final BufferAccumulator<Record<Event>> bufferAccumulator = BufferAccumulator.create(buffer, numberOfRecordsToAccumulate, bufferTimeout);
 
-        try (final ResponseInputStream<GetObjectResponse> object = s3Client.getObject(getObjectRequest)) {
-            codec.parse(object, record -> {
-                try {
-                    bufferAccumulator.add(record);
-                } catch (final Exception e) {
-                    LOG.error("Failed writing S3 objects to buffer.", e);
-                }
-            });
-        } catch (final Exception e) {
-            LOG.error("Error reading from S3 object: s3ObjectReference={}.", s3ObjectPointer, e);
-            s3ObjectsFailedCounter.increment();
-            throw e;
-        }
 
-        try {
-            bufferAccumulator.flush();
-        } catch (final Exception e) {
-            LOG.error("Failed writing S3 objects to buffer.", e);
-        }
+        s3ObjectReadTimer.wrap((Callable<Void>) () -> {
+            try (final ResponseInputStream<GetObjectResponse> object = s3Client.getObject(getObjectRequest)) {
+                codec.parse(object, record -> {
+                    try {
+                        bufferAccumulator.add(record);
+                    } catch (final Exception e) {
+                        LOG.error("Failed writing S3 objects to buffer.", e);
+                    }
+                });
+            } catch (final Exception e) {
+                LOG.error("Error reading from S3 object: s3ObjectReference={}.", s3ObjectPointer, e);
+                s3ObjectsFailedCounter.increment();
+                throw e;
+            }
+
+            try {
+                bufferAccumulator.flush();
+            } catch (final Exception e) {
+                LOG.error("Failed writing S3 objects to buffer.", e);
+            }
+
+            s3ObjectsSucceededCounter.increment();
+
+            return null;
```

Comment:
Do we need a return statement for this inline function?

Reply:
We do because we are using `Callable<Void>`. I can't use `Runnable` easily because it does not support checked exceptions. (The checked exception comes from `Buffer.writeAll()`).